File: app/src/main/java/com/kollectivemobile/euki/manager/ContentManager.java
package com.kollectivemobile.euki.manager;

import com.kollectivemobile.euki.model.ContentItem;
import com.kollectivemobile.euki.networking.EukiCallback;

import java.util.List;

public interface ContentManager {
    void getAbortionKnowledge(EukiCallback<ContentItem> callback);
    void getAbortionWalkthrough(EukiCallback<ContentItem> callback);
    void getSTIs(EukiCallback<ContentItem> callback);
    void getContraception(EukiCallback<ContentItem> callback);
    void getMiscarriage(EukiCallback<ContentItem> callback);
    void getSexRelationships(EukiCallback<ContentItem> callback);
    void getPregnancyOptions(EukiCallback<ContentItem> callback);
    void getMenstruationOptions(EukiCallback<ContentItem> callback);

    ContentItem getContentItem(String id);
    ContentItem getContentItem(String id, ContentItem contentItem);

    List<ContentItem> searchContentItem(String searchText);
    List<ContentItem> searchContentItem(String searchText, ContentItem contentItem);
}

File: app/src/main/java/com/kollectivemobile/euki/manager/ContentManager.kt
package com.kollectivemobile.euki.manager

import com.kollectivemobile.euki.model.ContentItem
import com.kollectivemobile.euki.networking.EukiCallback

interface ContentManager {
  fun getAbortionKnowledge(callback: EukiCallback<ContentItem>)
  fun getAbortionWalkthrough(callback: EukiCallback<ContentItem>)
  fun getSTIs(callback: EukiCallback<ContentItem>)
  fun getContraception(callback: EukiCallback<ContentItem>)
  fun getMiscarriage(callback: EukiCallback<ContentItem>)
  fun getSexRelationships(callback: EukiCallback<ContentItem>)
  fun getPregnancyOptions(callback: EukiCallback<ContentItem>)
  fun getMenstruationOptions(callback: EukiCallback<ContentItem>)

  fun getContentItem(id: String): ContentItem?
  fun getContentItem(id: String, contentItem: ContentItem): ContentItem?

  fun searchContentItem(searchText: String): List<ContentItem>
  fun searchContentItem(searchText: String, contentItem: ContentItem): List<ContentItem>
}
